### PR TITLE
Preserve content across retries during edit

### LINKED ISSUE #1593: The retry dialog drops unwritten changes
### Summary
When we `edit` a secret and it fails, the "Retry" dialog proposes to retry and re-open the editor, but it does so while dropping all the changes, which is a terrible UX.

### To reproduce 
```
gopass generate test 5
echo "testing" | gopass insert test user
gopass edit
```
This should display:
```
Password: Gv1rX
User: testing
```

Now edit it to look like:

```
Password: Gv1rX
User: testing
This is a nice body text without a new line between the key-value part and the body.
```
And save&exit your editor.
--> gopass complains about the secret not being in Native format (i.e. good Mime format) and proposes to "Retry: Y/n/q".
if you retry, it will re-open to:
```
Password: Gv1rX
User: testing
```
Having discarded all the changes.

This is really a bad UX, leading to frustration on the user side.

This is maybe related or maybe a duplicate of #1565 but triggered by another code path.

### PR DESCRIPTION
**Pull request by dominikschulz**
Fixes #1593

RELEASE_NOTES=[UX] Preserve content across retries

Signed-off-by: Dominik Schulz <dominik.schulz@gauner.org>

### CODE REVIEW

File: internal/action/edit.go [hunk 1]
```diff
@@ -46,26 +46,32 @@ func (s *Action) edit(ctx context.Context, c *cli.Context, name string) error {
...
 			return ExitError(ExitUnknown, err, "failed to invoke editor: %s", err)
 		}
+		tmpContent = newContent
 		if fromMIME {
-			nContent = append([]byte(secret.Ident+"\n"), nContent...)
-			if _, err := secparse.Parse(nContent); err != nil {
-				cont, err := termio.AskForBool(ctx, "ERROR: Can not parse Gopass native Secret. Retry?", true)
+			newContent = append([]byte(secret.Ident+"\n"), newContent...)
+			if _, err := secparse.Parse(newContent); err != nil {
+				out.Red(ctx, "ERROR: Can not parse Gopass native Secret: %s", err)
```
Comment (AnomalRoil) [inline internal/action/edit.go:61]: "cannot"

File: internal/action/edit.go [hunk 2]
```diff
@@ -46,26 +46,32 @@ func (s *Action) edit(ctx context.Context, c *cli.Context, name string) error {
...
+	tmpContent := content
 	for {
 		// invoke the editor to let the user edit the content
-		nContent, err := editor.Invoke(ctx, ed, content)
+		newContent, err := editor.Invoke(ctx, ed, tmpContent)
 		if err != nil {
 			return ExitError(ExitUnknown, err, "failed to invoke editor: %s", err)
 		}
+		tmpContent = newContent
 		if fromMIME {
```
Comment (AnomalRoil) [inline internal/action/edit.go:58]: What happens if it is not `fromMime` ?? Is it then not parsed at all?
We might want to do the same thing for insert wrt. #1602

File: internal/action/edit.go [hunk 3]
```diff
@@ -46,26 +46,32 @@ func (s *Action) edit(ctx context.Context, c *cli.Context, name string) error {
...
+				out.Red(ctx, "ERROR: Can not parse Gopass native Secret: %s", err)
+				out.Yellow(ctx, "Hint: Does your Key-Value section end with a new line?")
+				cont, err := termio.AskForBool(ctx, "Retry?", true)
 				if err != nil {
 					return err
 				}
 				if !cont {
 					return fmt.Errorf("malformed secret")
 				}
+				// retain edited content for next try
```
Comment (AnomalRoil) [inline internal/action/edit.go:70]: this could rather be a comment for line 57, no?

**Review by AnomalRoil — DISMISSED** [review]

File: internal/action/edit.go (hunk 2 above)
Comment (dominikschulz) [reply internal/action/edit.go:58]: Yes, if it's not fromMime it's not parsed at all.
And yes, we might want to do that for insert as well. But certainly in a different PR.

File: internal/action/edit.go (hunk 1 above)
Comment (dominikschulz) [reply internal/action/edit.go:61]: Done.

File: internal/action/edit.go (hunk 3 above)
Comment (dominikschulz) [reply internal/action/edit.go:70]: Uhm, yes. Indeed. I moved some things around.

File: internal/action/edit.go (hunk 2 above)
Comment (AnomalRoil) [reply internal/action/edit.go:59]: Agreed

**Review by AnomalRoil — APPROVED** [review]

**Pull request merged**